server/src/resources/vertipad/grpc.rs

```diff
@@ -4,74 +4,324 @@ mod grpc_server {
+    ///     id: "54acfe06-dd9b-42e8-8cb4-12a2fb2fa693"
+    /// }))
+    /// .await
+    /// {
+    ///    Ok(vertipad) => vertipad.into_inner(),
+    ///    Err(e) => panic!("Something went wrong retrieving the vertipad: {}", e),
+    /// };
+    /// ```
     async fn vertipad_by_id(&self, request: Request<Id>) -> Result<Response<Vertipad>, Status> {
         let id = request.into_inner().id;
-        match VERTIPADS.lock().await.get(&id) {
-            Some(vertipad) => Ok(Response::new(vertipad.clone())),
-            _ => Err(Status::not_found("Not found")),
+        if let Some(vertipad) = VERTIPADS.lock().await.get(&id) {
+            memdb_info!("Found entry for Vertipad. uuid: {}", id);
+            Ok(Response::new(vertipad.clone()))
+        } else {
+            let pool = get_psql_pool();
+            let data = VertipadPsql::new(&pool, Uuid::try_parse(&id).unwrap()).await;
+            match data {
+                Ok(vertipad) => Ok(Response::new(Vertipad {
+                    id,
+                    data: Some(vertipad.into()),
+                })),
+                Err(_) => Err(Status::not_found("Not found")),
+            }
         }
     }
 
-    ///vertipads // TODO implement. Currently returns arbitrary value
+    /// Takes a `SearchFilter` object to search the database with the provided values.
+    ///
+    /// This method supports paged results. When the `search_field` and `search_value` are empty, no filters will be applied.
+    ///
+    /// # Arguments
+    /// * self  - The VertipadRpc struct
+    /// * request - Request<SearchFilter> GRPC request
+    ///
+    /// # Returns
+    /// * Result Vertipads object or Status `Code::Internal` if there was an error in the search query
+    ///
+    /// # Example
+    /// ```
+    /// use svc_storage_client_grpc::client::vertipad_rpc_client::VertipadRpcClient;
+    /// use svc_storage_client_grpc::client::SearchFilter;
+    ///
+    /// let mut vertipad_client = VertipadRpcClient::connect(grpc_endpoint.clone()).await.unwrap();
+    /// let result = match vertipad_client.vertipads(tonic::Request::new(SearchFilter {
+    ///     search_field: "auth_method",
+    ///     search_value: "1",
+    ///     page_number: 1,
+    ///     results_per_page: 20
+    /// }))
+    /// .await
+    /// {
+    ///    Ok(vertipads) => vertipad.into_inner(),
+    ///    Err(e) => panic!("Something went wrong creating retrieving vertipads: {}", e),
+    /// };
+    /// ```
     async fn vertipads(
         &self,
-        _request: Request<SearchFilter>,
+        request: Request<SearchFilter>,
     ) -> Result<Response<Vertipads>, Status> {
-        let response = Vertipads {
-            vertipads: VERTIPADS.lock().await.values().cloned().collect::<_>(),
-        };
-        Ok(Response::new(response))
+        let filter = request.into_inner();
+        let mut filter_hash = HashMap::<String, String>::new();
+        filter_hash.insert("column".to_string(), filter.search_field);
```

Comment:
If I want to return all Vertipads/ports, in the client I have to specify search_field, search_value as `""`
Will this implementation return all vertipads or no vertipads?

Maybe we should add search_field search_value as optional?

Reply:
An empty search col/ value will return all vertipads.
I could make it optional, but I think a filter will be used pretty much every time.